libraries/common/components/InfiniteContainer/index.jsx

```diff
@@ -58,7 +60,13 @@ class InfiniteContainer extends Component {
      * 10ms was chosen because, on the one hand, it prevents the scroll event from flooding but,
      * on the other hand, it does not hinder users that scroll quickly from reloading next chunk.
      */
-    this.handleLoadingProxy = throttle(() => this.handleLoading(), 10);
+    this.handleLoadingProxy = throttle((force, nextProps) =>
+      (props.promiseBased
+        ? this.handleLoadingPromise(force, nextProps)
+        : this.handleLoading(force, nextProps)), 10);
+
+    this.handleLoadingProxy.bind(this);
+
```

Comment:
Here should be the same fix as in pwa6.

Reply:
I cherry picked the latest changes from PWA 6